lib/array_equals.rb

```diff
@@ -1,5 +1,37 @@
 # Determines if the two input arrays have the same count of elements
 # and the same integer values in the same exact order
+
 def array_equals(array1, array2)
-  raise NotImplementedError
+  # Check for nil
+  is_nil = true
+  if (array1 == nil) && (array2 == nil)
+    return is_nil
+  end
+  if (array1 == nil) && (array2 != nil)
+    is_nil = false
+    return is_nil
+  end
+  if (array1 != nil) && (array2 == nil)
+    is_nil = false
+    return is_nil
+  end
+
```

Comment:
The conditional statements on line 10 and line 14 can be combined into one statement using the `||` conditional.